### PR TITLE
#179 Consistently use URI object in APIs

### PR DESCRIPTION
**Pull request by ndoschek**
- Update to latest modelserver-client version
- Update model uri validation mechanism
- Provide wrapper hook to use validated model uri in interceptor handlers
  - Use hook for all default route providers
- Add default route provider for model element requests
- Clean up some duplicated types and extends route-utils with one generic query type and request handler
- Add README section about routing provider mechanism

Part of eclipse-emfcloud/emfcloud#179

Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: packages/modelserver-node/README.md [hunk 1]
```diff
@@ -5,3 +5,45 @@ This package provides the _Model Server_ server framework.
...
+```ts
+routerFactory('/api/v2/my-endpoint', myInterceptingHandler().bind(this));
+```
+
+This intercepting handler can:
+
+- handle the request by executing custom behaviour and not forwarding
+  - e.g. the [Validation RouteProvider](./src/routes/validation.ts)) delegates to the validation manager and stops the chain there
+- handle the request by executing custom behaviour and forwarding to the next upstream
+  - e.g. the [UndoRedo RouteProvider](./src/routes/undo-redo.ts)) delegates to the validation manager and stops the chain
```
Comment (eneufeld) [inline packages/modelserver-node/README.md:22]: This sentence seems to be wrong

File: packages/modelserver-node/src/routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * Copyright (c) 2022 STMicroelectronics.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ *******************************************************************************/
+export * from './route-utils';
```
Comment (eneufeld) [inline packages/modelserver-node/src/routes/index.ts:11]: is anybody using this?

**Review by eneufeld — APPROVED** [review]
Looks good.
There is a typing error in the commit message. 
See `READEME` instead of `README`

File: packages/modelserver-node/src/routes/index.ts (hunk 2 above)
Comment (ndoschek) [reply packages/modelserver-node/src/routes/index.ts:11]: Not yet, but if an adopter wants to re-use the validation wrapper hook, the import path is cleaner that way

**Pull request merged**